base/dcp_sharded.go

```diff
@@ -544,3 +549,33 @@ func (l *importHeartbeatListener) GetNodes() ([]string, error) {
 func (l *importHeartbeatListener) Stop() {
 	close(l.terminator)
 }
+
+// cbgtDestFactories map DCP feed keys (destKey) to a function that will generate cbgt.Dest.  Need to be stored in a
+// global map to avoid races between db creation and db addition to the server context database set
+
+type CbgtDestFactoryFunc = func() (cbgt.Dest, error)
+
+var cbgtDestFactories = make(map[string]CbgtDestFactoryFunc)
+var cbgtDestFactoriesLock sync.Mutex
+
+func StoreDestFactory(destKey string, dest CbgtDestFactoryFunc) {
+	cbgtDestFactoriesLock.Lock()
+	cbgtDestFactories[destKey] = dest
```

Comment:
Given it's an unexpected scenario that `destKey` already exists in the map, is it worth checking and returning an error from this function so it can be logged?